rpm/__init__.py

```diff
@@ -79,21 +93,77 @@ def try_path(path: str) -> bool:
     return False
 
 
+def import_helper(path: Path, suffixes: List) -> None:
+    """
+    Helps import rpm in cases where the shared object file extensions include the python
+    interpreter version
+
+    Args:
+        path (Path): the absolute path to the rpm module
+        suffixes (List): a list of extension suffixes valid for the python interpreter
+
+    Returns:
+        None
+    """
+    try:
+        importlib.reload(sys.modules[__name__])
+    except ModuleNotFoundError as e:
+        if e.name is None:
+            raise
+        logger.debug(
+            f"Failed to import {e.name} from {path}, "
+            "looking for extensions with valid suffixes"
+        )
+        stem = path / e.name.split(".")[1]
```

Comment:
I don't think this is necessary, we know the name of the module we are trying to import (`rpm._rpm`). Also, the plain `.so` extension is in the list of suffixes so it will be processed anyway.
I would probably merge this method with `try_path()`, as there shouldn't be a lot of code left in either method.

Reply:
What I have to do here is transform `rpm._rpm` from `e.name` to the file path `{sitepackages}/rpm/_rpm.cpython-39-x86_64-linux-gnu.so`. I could just assume that there will always be exactly one extension named `_rpm{suffix}` I guess? It would be safe in python 3.9 on RHEL 9. But for example, on older versions of the python bindings, there are other extensions like `_rpmb` and `_rpms`, and the assumption would fail in any such case. That doesn't seem safe to me. Is there another way to translate {e.name} into an expected extension path?

Reply:
as far as merging into try_path, can do, but I trying to avoid a nested try/except, just for legibility, like:

```
try:
    try:
        importlib.reload(sys.modules[__name__])
    except ModuleNotFounError as e:
       {...}
    confdir = 
finally:
```

Reply:
> What I have to do here is transform `rpm._rpm` from `e.name` to the file path `{sitepackages}/rpm/_rpm.cpython-39-x86_64-linux-gnu.so`. I could just assume that there will always be exactly one extension named `_rpm{suffix}` I guess? It would be safe in python 3.9 on RHEL 9. But for example, on older versions of the python bindings, there are other extensions like `_rpmb` and `_rpms`, and the assumption would fail in any such case. That doesn't seem safe to me. Is there another way to translate {e.name} into an expected extension path?

Oh, I see, I didn't fully think about what's going on. Importing `rpm` package fails as a result of failure to import the binary module. In that case your logic is correct, I don't think there is any other way but to get the module/file name from the exception. However there could be more binary modules (like you said, older versions had `_rpm`, `_rpmb` and `_rpms`, all imported from `rpm`), so it should be done in a loop. That also means that my `sys.modules[__name__] = module` suggestion is wrong, it has to be `sys.modules[module_name] = module`. And there should be one final `importlib.reload(sys.modules[__name__])` to reload `rpm` once the binary module(s) is/are loaded.

Reply:
Ok, makes sense. I will rework this a bit